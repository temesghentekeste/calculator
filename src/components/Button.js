import React from 'react';
import PropTypes from 'prop-types';

function Button({ name, clickHandler }) {
  return (
    <div>
      <button type="button" onClick={() => clickHandler(name)}>
        {name}
      </button>
    </div>
  );
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default Button;
