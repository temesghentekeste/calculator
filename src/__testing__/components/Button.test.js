import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Button from '../../components/Button';

let component;

beforeEach(() => {
  component = render(
    <Button name="+" clickHandler={handleClick} isLast={true} />
  );
});

const handleClick = () => {
  'Clicked';
};

it('renders the Button component', () => {
  component = renderer
    .create(<Button name="+" clickHandler={handleClick} isLast={true} />)
    .toJSON();

  expect(component).toMatchSnapshot();
});

it('renders the Button component with correct text content', () => {
  const { getByTestId } = render(
    <Button name="-" clickHandler={handleClick} isLast={true} />
  );

  expect(getByTestId('-')).toHaveTextContent('-');
});
