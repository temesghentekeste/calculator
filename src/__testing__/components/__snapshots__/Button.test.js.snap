// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Button component 1`] = `
<div
  className="calculatorButtonContainer calculatorButtonContainer_last"
>
  <button
    className="calculatorButtonContainer__button"
    data-testid="+"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
</div>
`;

exports[`renders the Button component 2`] = `
<div
  className="calculatorButtonContainer calculatorButtonContainer_last"
>
  <button
    className="calculatorButtonContainer__button"
    data-testid="+"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
</div>
`;
